lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -352,21 +355,45 @@ private void ScatterHelper(IList<T> messages, IList<NodeStruct<T>> order, int nu
         /// Checks if the identifier is registered with the Name Server.
         /// Throws exception if the operation fails more than the retry count.
         /// </summary>
-        /// <param name="identifier">The identifier to look up</param>
-        /// <param name="retries">The number of times to retry the lookup operation</param>
-        private void WaitForTaskRegistration(string identifier, int retries)
+        /// <param name="identifiers">The identifier to look up</param>
+        private void WaitForTaskRegistration(IList<string> identifiers)
         {
-            for (int i = 0; i < retries; i++)
+            using (Logger.LogFunction("OperatorTopology::WaitForTaskRegistration"))
             {
-                if (_nameClient.Lookup(identifier) != null)
+                IList<string> foundList = new List<string>();
+                try
+                {
+                    for (var i = 0; i < _retryCount; i++)
```

Comment:
That was original logic.  If we think applying retry on each registration makes more sense, then we don't need this change. 
As all the nodes are started almost at the same time and independently, they should be registered more or less at the same time. That is the reason I made this change. However, if we retry for the first node, by the time we check the second node, it should be registered already therefore there will be less sleep/retry anyway.